python-checks/src/main/java/org/sonar/python/checks/django/DjangoModelStringFieldCheck.java

```diff
@@ -0,0 +1,142 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.django;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.quickfix.PythonQuickFix;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.AssignmentStatement;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.ClassDef;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.quickfix.TextEditUtils;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6553")
+public class DjangoModelStringFieldCheck extends PythonSubscriptionCheck {
+
+  private static final String REPLACE_MESSAGE = "Replace this \"null=True\" flag with \"blank=True\".";
+  private static final String REMOVE_MESSAGE = "Remove this \"null=True\" flag.";
+  private static final String REPLACE_QUICK_FIX_MESSAGE = "Replace with \"blank=True\"";
+  private static final String REMOVE_QUICK_FIX_MESSAGE = "Remove \"null\" argument";
+
+  private static final String DJANGO_MODEL_FQN = "django.db.models.Model";
+  public static final Set<String> FIELD_TYPES_FQN = Set.of(
+    "django.db.models.CharField",
+    "django.db.models.TextField"
+  );
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, ctx -> {
+      var classDef = (ClassDef) ctx.syntaxNode();
+      if (TreeUtils.getParentClassesFQN(classDef).contains(DJANGO_MODEL_FQN)) {
+        classDef.body().statements().stream()
+          .filter(AssignmentStatement.class::isInstance)
+          .map(AssignmentStatement.class::cast)
+          .map(AssignmentStatement::assignedValue)
+          .filter(CallExpression.class::isInstance)
+          .map(CallExpression.class::cast)
+          .filter(DjangoModelStringFieldCheck::isTextField)
+          .forEach(call -> validateTextFieldArguments(ctx, call));
+      }
+    });
+  }
+
+  private static boolean isTextField(CallExpression call) {
+    return Optional.of(call)
+      .map(CallExpression::calleeSymbol)
+      .map(Symbol::fullyQualifiedName)
+      .filter(FIELD_TYPES_FQN::contains)
+      .isPresent();
+  }
+
+  private static void validateTextFieldArguments(SubscriptionContext ctx, CallExpression call) {
+    var nullArg = getCallArgumentByName(call, "null")
+      .filter(DjangoModelStringFieldCheck::isArgumentSetAsTrue);
+    var blankArg = getCallArgumentByName(call, "blank")
+      .filter(DjangoModelStringFieldCheck::isArgumentSetAsTrue);
+
```

Comment:
We can probably look for `unique=True` here and if it's present as well as `blankArg`, we can abort.

Reply:
done